src/main/java/io/confluent/connect/elasticsearch/ElasticsearchClient.java

```diff
@@ -154,6 +167,32 @@ public ElasticsearchClient(
         .build();
   }
 
+  /**
+   * Elastic High level Rest Client 7.17 has a compatibility mode to support ES 8. Checks the
+   * version number of ES to determine if we should be running in compatibility mode while using
+   * HLRC 7.17 to talk to ES.
+   */
+  private boolean shouldSetCompatibilityToES8() {
+    return !version().equals(UNKNOWN_VERSION_TAG)
+        && Integer.parseInt(version().split("\\.")[0]) >= 8;
+  }
+
+  private String getServerVersion(RestClient client) {
+    RestHighLevelClient highLevelClient = new RestHighLevelClientBuilder(client).build();
+    MainResponse response;
+    String esVersionNumber = UNKNOWN_VERSION_TAG;
+    try {
+      response = highLevelClient.info(RequestOptions.DEFAULT);
```

Comment:
highLevelClient.info itself internally will check useAPICompatibility ([here](https://github.com/elastic/elasticsearch/blob/312ddd6b6b41ce0dd516b7b3a197792ea3c77cf9/client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java#L1371-L1373)), 
just checking if `info` depends on useAPICompatibility flag in client itself.

Reply:
nit - there is a slight chance we may be checking only one ES server's version - [here](https://github.com/elastic/elasticsearch/blob/312ddd6b6b41ce0dd516b7b3a197792ea3c77cf9/client/rest/src/main/java/org/elasticsearch/client/RestClient.java#L296), the server corresponding to one of `connection.url`.

Reply:
QQ - what if we set ApiCompatibility = true by default for every request? would those requests be incompatible with ES version <7.11 ?

Reply:
> highLevelClient.info itself internally will check useAPICompatibility ([here](https://github.com/elastic/elasticsearch/blob/312ddd6b6b41ce0dd516b7b3a197792ea3c77cf9/client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java#L1371-L1373)),
just checking if info depends on useAPICompatibility flag in client itself.

From what I understand compatibility mode sends some extra headers so that ES8 recognises and works in ES7 mode. Not all calls requires the use of headers like the info call which is most likely the same between versions. 

> nit - there is a slight chance we may be checking only one ES server's version - [here](https://github.com/elastic/elasticsearch/blob/312ddd6b6b41ce0dd516b7b3a197792ea3c77cf9/client/rest/src/main/java/org/elasticsearch/client/RestClient.java#L296), the server corresponding to one of connection.url

I expect that if the customer is running ES in a cluster it won't be a mismatch of different ES versions. 

> QQ - what if we set ApiCompatibility = true by default for every request? would those requests be incompatible with ES version <7.11 

Elastic does not officially state that its incompatible though they do mention that it will work with 7.11 and above which says to me that older ones may not have been fully tested to work.